.github/workflows/preview.yml

```diff
@@ -0,0 +1,42 @@
+name: Preview
+on:
+  pull_request:
+    branches: 
+      - master
+      - develop
+      - 'release/**'
+jobs:
+  deployment:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      
+      - name: install deps
+        run: yarn
+
+      - name: build docs
+        run: yarn workspace @okta/design-docs build
+
+      - name: deploying docs to ods.dev
+        run: | 
+          SHA=$(git rev-parse --short ${{ github.event.pull_request.head.sha || github.sha }})
+          aws s3 sync ./packages/docs/dist/ s3://ods.dev/$SHA --delete
+      
+      - name: deployment complete
+        run: |
+          SHA=$(git rev-parse --short ${{ github.event.pull_request.head.sha || github.sha }})
+          curl \
+            -X POST \
+            $URL \
+            -H "Content-Type: application/json" \
+            -H "Authorization: token $GITHUB_TOKEN" \
+            --data '{ "body": "🤖 The latest preview of this branch is now available here: https://'"$SHA"'.ods.dev" }'
+        env:
+          URL: ${{ github.event.pull_request.comments_url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Spoke a bit with @brentmorris-okta - hopefully we can all reach agreement on whether or not this is a good idea.

Reply:
`secrets.GITHUB_TOKEN` is actually not something we explicitly set - it lives within actions. Still, I get your point. I think this would be most pertinent to `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as well as future items - for instance, a Slack incoming webhook URL.

Reply:
More on the topic here: https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets

Reply:
Spoke to @jmelberg-okta offline, he suggested that a question that may come up is around whether or not `secrets.GITHUB_TOKEN` has access to other repositories. 

Some quick research leads me to believe that this is not the case, which is in line with my original assumption. More on this below:

> When you enable GitHub Actions, GitHub installs a GitHub App on your repository. The GITHUB_TOKEN secret is a GitHub App installation access token. You can use the installation access token to authenticate on behalf of the GitHub App installed on your repository. The token’s permissions are limited to the repository that contains your workflow. For more information, see “Permissions for the `GITHUB_TOKEN`.” -- [Authenticating with the GITHUB_TOKEN](https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token)

> The permissions of GITHUB_TOKEN are limited to the repository that contains your workflow. -- [GitHub Support Community - Github Actions Token Scope](https://github.community/t/github-actions-token-scope/17061/2)

Reply:
🙏 for the due diligence here, y'all - both for our own assurance and giving guidance to other teams who may want something similar down the line.